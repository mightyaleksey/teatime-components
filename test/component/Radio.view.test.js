'use strict';

const Radio = require('../../component/Radio');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Radio.defaultProps.size).toBe('m');
});

test('defaultProp #theme should have a value `common`', () => {
  expect(Radio.defaultProps.theme).toBe('common');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Radio
      name='control'
      options={[
        {label: 'AA', value: 'aa'},
        {label: 'BB', value: 'bb'},
        {label: 'CC', value: 'cc'},
      ]}/>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Radio
      className='outer'
      disabled
      id='controls-id'
      name='control'
      options={[
        {label: 'AA', value: 'aa'},
        {label: 'BB', value: 'bb'},
        {label: 'CC', value: 'cc'},
      ]}
      theme='button'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
