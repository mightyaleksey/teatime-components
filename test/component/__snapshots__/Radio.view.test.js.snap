// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<div
  className="container"
  id={undefined}
  onChange={undefined}
>
  <span
    className="wrapper"
  >
    <input
      checked={false}
      className="native"
      defaultValue={undefined}
      disabled={undefined}
      id="_teatime2"
      name="control"
      onChange={[Function]}
      type="radio"
      value="aa"
    />
    <label
      className="control"
      htmlFor="_teatime2"
    />
    <label
      className="label"
      htmlFor="_teatime2"
    >
      AA
    </label>
  </span>
  <span
    className="wrapper"
  >
    <input
      checked={false}
      className="native"
      defaultValue={undefined}
      disabled={undefined}
      id="_teatime3"
      name="control"
      onChange={[Function]}
      type="radio"
      value="bb"
    />
    <label
      className="control"
      htmlFor="_teatime3"
    />
    <label
      className="label"
      htmlFor="_teatime3"
    >
      BB
    </label>
  </span>
  <span
    className="wrapper"
  >
    <input
      checked={false}
      className="native"
      defaultValue={undefined}
      disabled={undefined}
      id="_teatime4"
      name="control"
      onChange={[Function]}
      type="radio"
      value="cc"
    />
    <label
      className="control"
      htmlFor="_teatime4"
    />
    <label
      className="label"
      htmlFor="_teatime4"
    >
      CC
    </label>
  </span>
</div>
`;

exports[`Renders correctly with defined props 1`] = `
<div
  className="container outer"
  id="controls-id"
  onChange={undefined}
>
  <span
    className="wrapper"
  >
    <input
      checked={false}
      className="native"
      defaultValue={undefined}
      disabled={true}
      id="_teatime6"
      name="control"
      onChange={[Function]}
      type="radio"
      value="aa"
    />
    <label
      className="control"
      htmlFor="_teatime6"
    >
      AA
    </label>
  </span>
  <span
    className="wrapper"
  >
    <input
      checked={false}
      className="native"
      defaultValue={undefined}
      disabled={true}
      id="_teatime7"
      name="control"
      onChange={[Function]}
      type="radio"
      value="bb"
    />
    <label
      className="control"
      htmlFor="_teatime7"
    >
      BB
    </label>
  </span>
  <span
    className="wrapper"
  >
    <input
      checked={false}
      className="native"
      defaultValue={undefined}
      disabled={true}
      id="_teatime8"
      name="control"
      onChange={[Function]}
      type="radio"
      value="cc"
    />
    <label
      className="control"
      htmlFor="_teatime8"
    >
      CC
    </label>
  </span>
</div>
`;
